### ISSUE TITLE
Multi version panel filter same characters of version part from the title

### ISSUE THREAD

**Issue by pickalexrener** [MediaBrowser/Emby#3769]
### [Server version] 
4.8.0.19 beta

### [OS]
debian11

### [Problem]
When I try to rename my multi-version movies, it seems that emby will filter all the same characters from begining after ' - '. 
Here is how my files organized,
```
/media1/movie/jpn/Movie/House of Hummingbird (2019) [tmdbid=547404]/House of Hummingbird (2019) [tmdbid=547404] - 1080p_Wiki.mkv
/media1/movie/jpn/Movie/House of Hummingbird (2019) [tmdbid=547404]/House of Hummingbird (2019) [tmdbid=547404] - 1080p_HDC.mkv
```
In this case, the version selection panel will only show `Wiki` and `HDC` other than `1080p_Wiki` and `1080p_HDC` which quite annoying since I got a bunch of multi-version movies of same resolution.

I know if I changed to title like 'AAA - Wiki_1080p' / 'AAA - HDC_1080p' the selection panel will show 'Wiki_1080p' and 'HDC_1080p'. But still it doesn't make sense to me that emby identify different versions from the first different characters after '-'.  

From what I see in official [support](https://support.emby.media/support/solutions/articles/44001159102-movie-naming)

> If using the dash method _**anything following the dash**_ will be what you see in the Emby client app.

Clearly it dose not fit the situation above.

![version1](https://user-images.githubusercontent.com/51726771/221511439-c966fa92-69b8-4b59-8a21-61ec451b9634.png)

![version2](https://user-images.githubusercontent.com/51726771/221511551-02fe15e9-4e9d-4d07-84fc-6dd03ca59c09.png)

### [Update 1]
I looked up in Jellyfin repository, it seems quite simple when handling multi-version name — just remove the original folder name containing in the filename and do some trim actions. This is exactly what I expected emby should do.
 
https://github.com/jellyfin/jellyfin/blob/edc627fd5b5a5bd19c843dd9e2970b1ebce3fbfd/MediaBrowser.Controller/Entities/BaseItem.cs#L1168-L1235
```csharp
private string GetMediaSourceName(BaseItem item)
{
	var terms = new List<string>();

	var path = item.Path;
	if (item.IsFileProtocol && !string.IsNullOrEmpty(path))
	{
		if (HasLocalAlternateVersions)
		{
			var displayName = System.IO.Path.GetFileNameWithoutExtension(path)
				.Replace(System.IO.Path.GetFileName(ContainingFolderPath), string.Empty, StringComparison.OrdinalIgnoreCase)
				.TrimStart(new char[] { ' ', '-' });

			if (!string.IsNullOrEmpty(displayName))
			{
				terms.Add(displayName);
			}
		}

		if (terms.Count == 0)
		{
			var displayName = System.IO.Path.GetFileNameWithoutExtension(path);
			terms.Add(displayName);
		}
	}

	if (terms.Count == 0)
	{
		terms.Add(item.Name);
	}

...some extra things to do 


	return string.Join('/', terms);
}
```

### [Update 2]
I decompiled the MediaBrowser.Controller.dll in 4.8.0.19 beta release, compare it with similar code fragment in Jellyfin and find out that emby doing extra work of trims.

**_GetMediaSources_** in class MediaBrowser.Controller.Entites.BaseItem of **Jellyfin**
```csharp
public virtual List<MediaSourceInfo> GetMediaSources(bool enablePathSubstitution)
{
	if (SourceType == SourceType.Channel)
	{
		var sources = ChannelManager.GetStaticMediaSources(this, CancellationToken.None)
			.ToList();

		if (sources.Count > 0)
		{
			return sources;
		}
	}

	var list = GetAllItemsForMediaSources();
	var result = list.Select(i = > GetVersionInfo(enablePathSubstitution, i.Item, i.MediaSourceType)).ToList();

	if (IsActiveRecording())
	{
		foreach(var mediaSource in result)
		{
			mediaSource.Type = MediaSourceType.Placeholder;
		}
	}

	return result.OrderBy(i = >
	{
		if (i.VideoType == VideoType.VideoFile)
		{
			return 0;
		}

		return 1;
	}).ThenBy(i = > i.Video3DFormat.HasValue ? 1 : 0)
		.ThenByDescending(i = >
	{
		var stream = i.VideoStream;

		return stream is null || stream.Width is null ? 0 : stream.Width.Value;
	})
		.ToList();
}
```

**_GetMediaSource_** in class MediaBrowser.Controller.Entites.BaseItem of **Emby**:
```csharp
public virtual List<MediaSourceInfo> GetMediaSources(User user, bool enableAlternateMediaSources, bool enablePathSubstitution, LibraryOptions libraryOptions)
{
	List<MediaSourceInfo> list = (from i in this.GetAllItemsForMediaSources(user, enableAlternateMediaSources)
		select BaseItem.GetVersionInfo(enablePathSubstitution, i.Item1, i.Item2, libraryOptions)).ToList<MediaSourceInfo>();
	if (this.IsActiveRecording())
	{
		foreach(MediaSourceInfo mediaSourceInfo in list)
		{
			mediaSourceInfo.Type = MediaSourceType.Placeholder;
		}
	}
	if (list.Count > 1)
	{
		this.TrimCommonNames(list); // this is where emby does extra trim and confused me
	}
	return list.OrderByDescending(delegate(MediaSourceInfo i)
	{
		MediaStream videoStream = i.VideoStream;
		if (videoStream != null && videoStream.Width != null)
		{
			return videoStream.Width.Value;
		}
		return 0;
	}).ToList<MediaSourceInfo>();
}
```

TrimCommonNames:

```csharp
private string[] TrimCommonNames(string[] names)
{
	string text = new string(names.First<string>().Substring(0, names.Min((string s) = > s.Length)).TakeWhile((char c, int i) = > names.All((string s) = > s[i] == c)).ToArray<char>());
	int commonPrefixLength = text.Length;
	if (commonPrefixLength < 1)
	{
		return names;
	}
	foreach(string text2 in names)
	{
		if (text2.Length <= commonPrefixLength)
		{
			return names;
		}
		int num = commonPrefixLength;
		char c2 = text2[num];
		while (!this.IsNameSplitChar(c2) && num > 0)
		{
			num--;
			c2 = text2[num];
		}
		commonPrefixLength = Math.Max(num, 0);
	}
	if (commonPrefixLength < 1)
	{
		return names;
	}
	string[] source = (from i in names
		select i.Substring(commonPrefixLength).Trim(BaseItem.MediaSourceNameTrimChars)).ToArray<string>();
	if (source.Any((string i) = > i.Length < 1))
	{
		return names;
	}
	return source.Select(new Func<string, string>(this.NormalizeName)).ToArray<string>();
}
```

I'm not very good at reading c#, but the **_TrimCommonNames_** function clearly explains why emby multi-verison drop pannel not working as I expected. So is this a feature or bug? Do I need to pull another issue about feature request or start a discussion about whether it's necessary to trim common names? 
I still insist that drop panel should show exactly what was written in the original filename, no need to do the  common names trim.

### [Suggestion]
Keep selection drop panel's value the same as local file's format.
File:
```
/media1/movie/jpn/Movie/House of Hummingbird (2019) [tmdbid=547404]/House of Hummingbird (2019) [tmdbid=547404] - 1080p_Wiki.mkv
/media1/movie/jpn/Movie/House of Hummingbird (2019) [tmdbid=547404]/House of Hummingbird (2019) [tmdbid=547404] - 1080p_HDC.mkv
```
Drop panel 
```
1080p_Wiki
1080p_HDC
```

**Comment by pickalexrener** [comment]
### [Temporary Solution]
I decompiled MediaBrowser.Controller.dll and delete the segement from MediaBrowser.Controller.Entites.BaseItem metioned above. 
```csharp
	if (list.Count > 1)
	{
		this.TrimCommonNames(list); // this is where emby does extra trim and confused me
	}
```

Save - compile - backup - replace and finally it works as I expected.
![Snipaste_2023-03-01_08-05-42](https://user-images.githubusercontent.com/51726771/222013302-e931b1a4-a37a-44a9-b22e-d48bf4f831f1.png)

**Comment by LukePulverenti** [comment]
Hi, this will be resolved in the upcoming 4.8 server release. Thanks.
